entity/dto/pet_insert_dto.go

```diff
@@ -0,0 +1,46 @@
+package dto
+
+import (
+	"errors"
+	"pet-dex-backend/v2/pkg/uniqueEntityId"
+	"regexp"
+	"time"
+)
+
+var notEmptyRegex = regexp.MustCompile(`^\S+$`)
+
+var nameRegex = regexp.MustCompile(`^[a-zA-Z0-9\s]+$`)
+
+var sizeRegex = regexp.MustCompile(`^(small|medium|large|giant)$`)
+
+type PetInsertDto struct {
+	Name         string          `json:"name"`
+	UserID       uniqueEntityId.ID `json:"user_id"`
+	BreedID      uniqueEntityId.ID `json:"breed_id"`
+	AdoptionDate *time.Time      `json:"adoption_date"`
+	Birthdate    *time.Time      `json:"birthdate"`
+	Weight       float64         `json:"weight"`
+	Size         string          `json:"size"`
+}
+
+func (p *PetInsertDto) Validate() error {
+	if !nameRegex.MatchString(p.Name) {
+		return errors.New("name cannot be empty")
+	}
+	if len(p.Name) > 80 {
+		return errors.New("name cannot exceed 80 characters")
```

Comment:
bom verificar se o a string é vazia também

```suggestion
	if p.Name == "" || len(p.Name) > 80 {
```

Reply:
Adicionei a verificação logo na linha debaixo, coloquei em outro if para manter as validações isoladas